packages/next/export/worker.ts

```diff
@@ -420,14 +422,20 @@ export default async function exportPage({
         if (optimizeCss) {
           process.env.__NEXT_OPTIMIZE_CSS = JSON.stringify(true)
         }
-        renderResult = await renderMethod(
-          req,
-          res,
-          page,
-          query,
-          // @ts-ignore
-          curRenderOpts
-        )
+        try {
+          renderResult = await renderMethod(
+            req,
+            res,
+            page,
+            query,
+            // @ts-ignore
+            curRenderOpts
+          )
+        } catch (err: any) {
+          if (err.digest !== NEXT_DYNAMIC_NO_SSR_CODE) {
```

Comment:
It seems `NEXT_DYNAMIC_NO_SSR_CODE` fully overlaps with something @hanneslund has been working on to bail out of SSR, would be good to leverage that instead: https://github.com/vercel/next.js/pull/43603/files#diff-5becfaf098605c339c89c458fa973f977de17d9176b801680a04a92e91b340cbR35

Reply:
but that seems not being handled on client? like for `onRecoverableError`

Reply:
Probably should be!

Reply:
yeah I think I can try to merge them together later!